### PR TITLE
feat(MoveInPlace): add controller rotation control method

### PR DESCRIPTION
**Pull request by kjack9**
Add an additional method to direct locomotion for Motion In Place.

Controller Direction will average the rotation/direction of the two controllers and use that value for movement.  If only one controller is present, only that one value will be
used.

This is my first PR to VRTK, so please let me know if I need to change anything stylistically.

### CODE REVIEW

**Comment by thestonefox** [comment]
Update the commit message with `git commit --amend` (then force push)

so:

 * it's within 72 characters per line
 * the subject line feature can now be `(MoveInPlace)` as it's a known script
 * the subject line should be all lower case (after the colon)
 * add a line break between the subject line and body
 * also no need to include the bit about being your first PR

It should read like this:

```
feat(MoveInPlace): add controller rotation control method

Add an additional method to direct locomotion for Motion In Place.

Controller Direction will average the rotation/direction of the two
controllers and use that value for movement. If only one controller
is present, only that one value will be used.
```

**Comment by kjack9** [comment]
Updated as requested.

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 1]
```diff
@@ -312,6 +314,10 @@ private void FixedUpdate()
                     }
                     direction = initalGaze;
                 }
+                // if we're doing controller rotation movement
+                else if (directionMethod.Equals(DirectionalMethod.ControllerRotation)) {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 2]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
             rigidBody.MovePosition(rigidBody.transform.position + movement);
         }
 
+        private Quaternion determineAverageControllerRotation() {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: Uppercase Method name should be `DetermineAverageControllerRotation`

bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 3]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+            // Build the average rotation of the controller(s)
+            Quaternion newRotation;
+
+
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: remove additional line break

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 4]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+            // Both controllers are present
+            if (controllerLeftHand != null && controllerRightHand != null) {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 5]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+                newRotation = averageRotation(controllerLeftHand.transform.rotation, controllerRightHand.transform.rotation);
+            }
+            // Left controller only
+            else if (controllerRightHand != null && controllerRightHand == null) {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 6]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+                newRotation = controllerLeftHand.transform.rotation;
+            }
+            // Right controller only
+            else if (controllerRightHand != null && controllerLeftHand == null) {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 7]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+                newRotation = controllerRightHand.transform.rotation;
+            }
+            // No controllers!
+            else {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 8]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+                newRotation = Quaternion.identity;
+            }
+
+            return newRotation;
+        }
+
+        // Returns the average of two Quaternions
+        private Quaternion averageRotation(Quaternion rot1, Quaternion rot2) {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: Uppercase method name should be `AverageRotation`

bracket on new line like

```
func
{
}
```

File: Assets/VRTK/Scripts/VRTK_MoveInPlace.cs [hunk 9]
```diff
@@ -333,6 +339,41 @@ private void FixedUpdate()
...
+            return Quaternion.Slerp(rot1, rot2, 0.5f);
+        }
+
+        // Returns a Vector3 with only the X and Z components (Y is 0'd)
+        private static Vector3 vector3XZOnly(Vector3 vec) {
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_MoveInPlace.cs]: Uppercase method name should be Vector3XZOnly

bracket on new line like

```
func
{
}
```

**Comment by thestonefox** [comment]
Tested it works really well ❤️ 💚 💛 💙

**Pull request merged**